### PR TITLE
Frontend and Backend API SingIn and SignUp

### PR DESCRIPTION
**Pull request by rnavaneeth992**
Issue: #98 

Added files

### CODE REVIEW

File: src/app/layout.js
Comment (LifeAsDev) [inline src/app/layout.js:1]: you delete the MuiThemeProvider

Comment (LifeAsDev) [reply src/app/layout.js:1]: and also why you comment the metadata?

File: src/components/right/RigthSignUp.jsx [hunk 1]
```diff
@@ -81,228 +105,219 @@ const RigthSignUp = () => {
...
+  };
+
+  const closeModal = () => {
+    setModalContent(null);
+    // Reset mutation state
+    registerUserMutation.reset();
   };
 
-  const [isMobile, setIsMobile] = useState(false);
+  const [isMobile, setIsMobile] = useState(window.innerWidth <= 767);
```
Comment (LifeAsDev) [inline src/components/right/RigthSignUp.jsx]: warning, window is not defined

**Review by LifeAsDev — COMMENTED** [review]
review your code

**Comment by rnavaneeth992** [comment]
MuIthemeprovider pushed

Metadata is not working if i am working with react-query 

window warning is not coming for me, even though i added some validation please check it @LifeAsDev @akshayw1

**Comment by LifeAsDev** [comment]
> MuIthemeprovider pushed
> Great
> Metadata is not working if i am working with react-query
> Can you explain me that issue?
> window warning is not coming for me, even though i added some validation please check it @LifeAsDev @akshayw1
yeah I know, but I fixed that warning, you have a old file solve that, just set it to false

**Comment by rnavaneeth992** [comment]
> > MuIthemeprovider pushed
> > Great
> > Metadata is not working if i am working with react-query
> > Can you explain me that issue?
> > window warning is not coming for me, even though i added some validation please check it @LifeAsDev @akshayw1
> > yeah I know, but I fixed that warning, you have a old file solve that, just set it to false (Oh Thanks)

Metadata if i add (it is telling you can't use in 'use client' if i remove and add then react-query I am unable to use it that's why i removed

**Comment by LifeAsDev** [comment]
I dont notice you add 'use client' in layout, Dont add 'use client' in layout that is bad

**Comment by LifeAsDev** [comment]
create a provider for query like the muithemeprovider and delete 'use client' of layout

**Comment by rnavaneeth992** [comment]
> create a provider for query like the muithemeprovider and delete 'use client' of layout

OH OK Bro got it I will do it and update you

File: src/components/right/RigthSignUp.jsx [hunk 2]
```diff
@@ -81,228 +105,231 @@ const RigthSignUp = () => {
...
+    registerUserMutation.reset();
+  };
+
+  const [isMobile, setIsMobile] = useState(() => {
+    // Use window.innerWidth only on the client side
+    if (typeof window !== 'undefined') {
+      return window.innerWidth <= 767;
+    }
+    return false;
+  });
```
Comment (LifeAsDev) [inline src/components/right/RigthSignUp.jsx:199]: adsad

**Comment by rnavaneeth992** [comment]
> create a provider for query like the muithemeprovider and delete 'use client' of layout

@LifeAsDev I tried even though some different errors are coming 

Uncaught Error: invariant expected app router to be mounted
    at useRouter (navigation.js:125:15)
    at HotReload (hot-reloader-client.js:355:46)
    at renderWithHooks (react-dom.development.js:11009:18)
    at mountIndeterminateComponent (react-dom.development.js:16761:13)
    at beginWork$1 (react-dom.development.js:18345:16)
    at beginWork (react-dom.development.js:26741:14)
    at performUnitOfWork (react-dom.development.js:25587:12)
    at workLoopSync (react-dom.development.js:25303:5)
    at renderRootSync (react-dom.development.js:25258:7)
    at recoverFromConcurrentError (react-dom.development.js:24475:20)
    at performConcurrentWorkOnRoot (react-dom.development.js:24420:26)
    at workLoop (scheduler.development.js:261:34)
    at flushWork (scheduler.development.js:230:14)
    at MessagePort.performWorkUntilDeadline (scheduler.development.js:534:21)
    
 Even though I tried like Muithemeprovider 
 
 Can you tell how to do it

**Comment by LifeAsDev** [comment]
do a commit, I will clone your repo

**Comment by rnavaneeth992** [comment]
@LifeAsDev Done can you check please

**Comment by LifeAsDev** [comment]
you forget to add the children prop

```

'use client'
import { QueryClient, QueryClientProvider } from 'react-query';
const queryClient = new QueryClient();

const QueryProvider = () => {
    return (
        <QueryClientProvider client={queryClient}>
        </QueryClientProvider>
    );
  };
  export default QueryProvider;

```

**Comment by LifeAsDev** [comment]
Look the children prop here

```
const MuiThemeProvider = ({ children }) => {
  return <ThemeProvider theme={theme}>{children} </ThemeProvider>;
};
```

**Comment by LifeAsDev** [comment]
also dont wrap html and body with the providers only the children

```
export default function RootLayout({ children }) {
  return (
    <html lang="en">
      <body className={inter.className}>
        <QueryClientProvider client={queryClient}>
          <AuthProvider>
            <MuiThemeProvider>{children}</MuiThemeProvider>
          </AuthProvider>
        </QueryClientProvider>
      </body>
    </html>
  );
}

```

**Comment by LifeAsDev** [comment]
Also you did not import the QueryProvider that you created on layout instead you imported the QueryClientProvider

**Comment by rnavaneeth992** [comment]
@LifeAsDev I pushed changes now, Can you check

**Comment by LifeAsDev** [comment]
> @LifeAsDev I pushed changes now, Can you check

yeah I see, is it working without errors?

**Comment by rnavaneeth992** [comment]
> > @LifeAsDev I pushed changes now, Can you check
> 
> yeah I see, is it working without errors?

Ya it is working 
image

**Comment by rnavaneeth992** [comment]
@LifeAsDev @akshayw1 Can you check and merge it please

**Pull request merged**